### PR TITLE
Add SelectField wrapper

### PR DESCRIPTION
**Pull request by bkjohnson**
Closes https://github.com/department-of-veterans-affairs/formulate/issues/10

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "tsdx": "^0.14.1",
     "tslib": "^2.3.0",
     "typescript": "^4.3.5",
-    "web-components": "department-of-veterans-affairs/component-library#wc-v0.9.0"
+    "web-components": "https://github.com/department-of-veterans-affairs/component-library.git#wc-bindings-packaging"
```
Comment (bkjohnson) [inline package.json:50]: We'll want to make sure these component changes get released before this `formulate` PR gets merged.

File: jest.config.js [hunk 2]
```diff
@@ -175,10 +175,7 @@ module.exports = {
   // transform: undefined,
 
   // An array of regexp pattern strings that are matched against all source file paths, matched files will skip transformation
-  // transformIgnorePatterns: [
-  //   "/node_modules/",
-  //   "\\.pnp\\.[^\\/]+$"
-  // ],
+  transformIgnorePatterns: ['/node_modules/^(web-components)'],
```
Comment (bkjohnson) [inline jest.config.js:178]: With the introduction of the React bindings, the import was breaking the Jest test. This ensures that the bindings get transformed through babel.

Comment (cvalarida) [reply jest.config.js:178]: :thinking: Ideally, the result of the web components React bindings build doesn't need to be transpiled in any way. We may need to fix this upstream.

File: package.json [hunk 3]
```diff
@@ -47,7 +47,7 @@
     "tsdx": "^0.14.1",
     "tslib": "^2.3.0",
     "typescript": "^4.3.5",
-    "web-components": "department-of-veterans-affairs/component-library#wc-v0.9.0"
+    "web-components": "https://github.com/department-of-veterans-affairs/component-library.git#wc-v0.9.3"
```
Comment (cvalarida) [inline package.json:50]: I prefer the shorter version, but :man_shrugging:

File: src/utils/validation.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { FieldProps } from '../form-builder/types';
+import { getMessage } from './i18n';
+
+export const validator = (
```
Comment (cvalarida) [inline src/utils/validation.ts:4]: This is great. Thanks for moving it!

File: src/form-builder/SelectField.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import { useField, FieldHookConfig } from 'formik';
+
+import { FieldProps } from './types';
+import { validator } from '../utils/validation';
+
+import { VaSelect } from 'web-components/react-bindings';
+
+type SelectProps = FieldProps<string> & {
+  onVaSelect: (e: CustomEvent) => void;
+  children: HTMLOptionElement[];
```
Comment (cvalarida) [inline src/form-builder/SelectField.tsx:11]: :100:

File: test/form-builder/SelectField.test.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { waitFor } from '@testing-library/react';
+
+import SelectField from '../../src/form-builder/SelectField';
+import { buildRenderForm, changeValue } from '../utils';
+
+const renderForm = buildRenderForm({});
+
+const getInput = (container: HTMLElement): HTMLInputElement => {
```
Comment (cvalarida) [inline test/form-builder/SelectField.test.tsx:9]: :thinking: Can we import the actual type from `web-components` now?

If not, should this be [`HTMLSelectElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement)?

File: test/form-builder/SelectField.test.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    });
+    expect(input.getAttribute('error')).toEqual('Please provide a response');
+  });
+
+  test('renders a custom "required" validation error message', async () => {
+    const { container, getFormProps } = renderForm(
+      <SelectField
+        name="thing"
+        label="The Thing"
+        required="You need to fill this in, bub"
```
Comment (cvalarida) [inline test/form-builder/SelectField.test.tsx:53]: :grin:

**Review by cvalarida — APPROVED** [review]
Looks good. I think we'll need to reassess the output of those React bindings, but that's definitely out of scope for this PR. This is great. :+1:

File: test/form-builder/SelectField.test.tsx (hunk 7 above)
Comment (bkjohnson) [reply test/form-builder/SelectField.test.tsx:53]: :spaghetti:

File: jest.config.js (hunk 2 above)
Comment (bkjohnson) [reply jest.config.js:178]: For more context, here is the problem that this fixes:

image

I tried a few different things, like specifying a `transform` key for `babel-jest` but that didn't work.

I don't fully understand why Jest is complaining about an import in `node_modules/web-components/react-bindings/index.js` :shrug:

File: test/form-builder/SelectField.test.tsx (hunk 6 above)
Comment (bkjohnson) [reply test/form-builder/SelectField.test.tsx:9]: Good call, I'll see if that can be changed.

Comment (bkjohnson) [reply test/form-builder/SelectField.test.tsx]: https://github.com/department-of-veterans-affairs/formulate/pull/16/commits/15ea070003722fdf1985a3d44e6cada6345ecefa

**Pull request merged**